FV3/gfsphysics/GFS_layer/GFS_physics_driver.F90

```diff
@@ -1502,11 +1513,21 @@ subroutine GFS_physics_driver                                 &
 !       ' adjsfculw3=',adjsfculw3(ipr,:),' icefr=',Sfcprop%fice(ipr),' tsfc3=',tsfc3(ipr,:)
 !
         do i=1,im
-          Diag%dlwsfc(i) = Diag%dlwsfc(i) +   adjsfcdlw(i)*dtf
+          Diag%dlwsfc(i) = Diag%dlwsfc(i) +   adjsfcdlw_for_lsm(i)*dtf
           Diag%ulwsfc(i) = Diag%ulwsfc(i) +   adjsfculw(i)*dtf
           Diag%psmean(i) = Diag%psmean(i) + Statein%pgr(i)*dtf        ! mean surface pressure
         enddo
 
+        if (Model%override_surface_radiative_fluxes) then
+          do i=1,im
+            Diag%dswsfc(i) = Diag%dswsfc(i) + adjsfcdsw_for_lsm(i)*dtf
+            Diag%uswsfc(i) = Diag%uswsfc(i) + (adjsfcdsw_for_lsm(i) - adjsfcnsw_for_lsm(i))*dtf
+
+            Diag%dlwsfc_rrtmg(i) = Diag%dlwsfc_rrtmg(i) + adjsfcdlw(i)*dtf
+            Diag%ulwsfc_rrtmg(i) = Diag%ulwsfc_rrtmg(i) + adjsfculw(i)*dtf
```

Comment:
why don't the shortwave rrtmg diagnostics get updated here?

Reply:
oh, is it because you're just using the `fluxr` array to hold those?

Reply:
Indeed this is confusing, but is the way the model currently works.  In the normal (and by extension, RRTMG) case the accumulation buckets for the time-mean surface shortwave radiative fluxes are incremented in the radiation driver (not the physics driver):

https://github.com/VulcanClimateModeling/fv3gfs-fortran/blob/bd1fde4d92788cc1a39d4152a3d00121f9a70439/FV3/gfsphysics/GFS_layer/GFS_radiation_driver.F90#L2084-L2085

Reply:
Yeah, if there's an implementation reason why the shortwave and longwave time-averaging are handled differently it might be good to document that somewhere

Reply:
I'm afraid I do not know the big picture reason why things were implemented this way originally.  What is also confusing to me is that the instantaneous and time-mean shortwave diagnostics are handled in different places.  E.g. here in the physics driver for the instantaneous upward shortwave radiative flux at the surface:
https://github.com/VulcanClimateModeling/fv3gfs-fortran/blob/bd1fde4d92788cc1a39d4152a3d00121f9a70439/FV3/gfsphysics/GFS_layer/GFS_physics_driver.F90#L2079

versus here in the radiation driver for the time-mean:

https://github.com/VulcanClimateModeling/fv3gfs-fortran/blob/bd1fde4d92788cc1a39d4152a3d00121f9a70439/FV3/gfsphysics/GFS_layer/GFS_radiation_driver.F90#L2084

I discuss this some towards the end of [this section in my Dropbox paper doc](https://paper.dropbox.com/doc/2021-02-17-adding-a-surface-radiative-flux-bias-correction-to-FV3GFS--BGuxmBKY3tbqNsg~R7g8rdfpAg-JFh8VUrFlcGY78AXKibUy#:uid=476513069356341896423152&h2=How-do-these-radiation-inputs-).

Reply:
Ah okay thanks. Well I think it's fine to leave the PR as is given you are just following the current implementation.

Reply:
Ok sounds good -- I'll go ahead and merge.